pennylane/measurements/mid_measure.py

```diff
@@ -516,3 +516,19 @@ def __str__(self):
 
     def __repr__(self):
         return f"MeasurementValue(wires={self.wires.tolist()})"
+
+
+def find_measured_mcms(circuit):
```

Comment:
I'm curious about the decision to place this function here.

Is this going to be part of the API for the measurements module? If so, do we need to import it and document it in `measurements/__init__.py`?

Reply:
> Is this going to be part of the API for the measurements module? 

I don't think so, or at least I had not envisioned it.

I'm not really sure where to put it. Could be there since it is an MCM-related utility function. I'm thinking using it to improve `dynamic_one_shot`'s performance too, so it could be in that transform's module as well. Or just `DQ::simulate` for now ...